x-pack/plugins/observability/server/services/slo/get_slo.ts

```diff
@@ -5,24 +5,23 @@
  * 2.0.
  */
 
-import { GetSLOResponse, getSLOResponseSchema } from '@kbn/slo-schema';
-import { SLO, SLOId, SLOWithSummary, Summary } from '../../domain/models';
+import { ALL_VALUE, GetSLOParams, GetSLOResponse, getSLOResponseSchema } from '@kbn/slo-schema';
+import { SLO, Summary } from '../../domain/models';
 import { SLORepository } from './slo_repository';
 import { SummaryClient } from './summary_client';
 
 export class GetSLO {
   constructor(private repository: SLORepository, private summaryClient: SummaryClient) {}
 
-  public async execute(sloId: string): Promise<GetSLOResponse> {
+  public async execute(sloId: string, params: GetSLOParams = {}): Promise<GetSLOResponse> {
     const slo = await this.repository.findById(sloId);
-    const summaryBySlo = await this.summaryClient.fetchSummary([slo]);
+    const instanceId = params.instanceId ?? ALL_VALUE;
+    const summary = await this.summaryClient.computeSummary(slo, instanceId);
```

Comment:
💬 Compute summary was used only with array of 1 element, so I refactored it to take just an SLO, with an optional instanceId.